index.js

```diff
@@ -9,92 +9,107 @@ const port = process.env.PORT || 5000;
+    secret: "Tell me what thy lordly name is on the Night's Plutonian shore",
     cookie: {
-        maxAge: 1 * 24 * 60 * 60 * 1000,
-        secure: false, // Would be true to restrict to https, but cannot test with Postman while set to true.
+      maxAge: 1 * 24 * 60 * 60 * 1000,
+      secure: false, // Would be true to restrict to https, but cannot test with Postman while set to true.
     },
     httpOnly: true,
     resave: false,
     saveUninitialized: false,
-}));
+  })
+);
 const corsOptions = {
-    credentials: true,
-    origin: 'http://localhost:3000', // Obviously changed for prod
-}
+  credentials: true,
+  origin: 'http://localhost:3000', // Obviously changed for prod
+};
 server.use(cors(corsOptions));
 server.use(express.json());
 server.use('/api/restricted', restrictAccess);
 
-function restrictAccess(req, res, next){
-    if (req.session && req.session.username) next()
-    else res.status(401).json({ message: 'You shall not pass!' });
+function restrictAccess(req, res, next) {
+  if (req.session && req.session.username) next();
+  else res.status(401).json({ message: 'You shall not pass!' });
 }
 
-server.post('/api/register', (req, res) => { // Promise syntax
-    const { username, password, email, firstName, lastName } = req.body;
-    if (!username || !password || !email || !firstName || !lastName) 
-        return res.status(400).json({ error: 'Please provide values for username, password, email, firstName, and lastName keys.' });
-    const credentials = { username, password, email, firstName, lastName };
-    const hash = bcrypt.hashSync(credentials.password, 14);
-    credentials.password = hash;
-    db('users')
-        .insert(credentials)
-        .then(id => {
-            req.session.username = username;
-            res.status(201).json({ id: id[0] });
-        })
-        .catch(error => {
-            const column = error.message.match(/users\..+/)[0];
-            const reason = error.message.match(/\w+(?=\sconstraint\sfailed)/)[0];
-            switch(reason){
-                case 'UNIQUE':
-                    return res.status(400).json({ error: `Value provided for ${column} already exists.`});
-                default:
-                    return res.status(500).json({ error });
-            }
-        });
+server.post('/api/register', (req, res) => {
+  // Promise syntax
+  const { username, password, email, firstName, lastName } = req.body;
+  if (!username || !password || !email || !firstName || !lastName)
+    return res.status(400).json({
+      error:
+        'Please provide values for username, password, email, firstName, and lastName keys.',
+    });
+  const credentials = { username, password, email, firstName, lastName };
+  const hash = bcrypt.hashSync(credentials.password, 14);
+  credentials.password = hash;
+  db('users')
+    .insert(credentials)
+    .then(id => {
+      req.session.username = username;
+      res.status(201).json({ id: id[0] });
```

Comment:
You could use [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring) here:
```javascript
.then(([ id ]) => {
  req.session.username = username;
  res.status(201).json({ id });
})
```
since the parameter name `id` doesn't really imply an array. You could also just call `id` `ids` instead for clarity.